core/templates/dev/head/pages/collection_editor/settings_tab/CollectionPermissionsCardDirective.js

```diff
@@ -22,10 +22,24 @@ oppia.directive('collectionPermissionsCard', [function() {
     restrict: 'E',
     templateUrl: 'inline/collection_permissions_card_directive',
     controller: [
-      '$scope',
-      function($scope) {
-        $scope.ownerNames = GLOBALS.ownerNames;
-        $scope.isPrivate = GLOBALS.isPrivate;
+      '$scope', 'CollectionRightsObjectFactory',
+      'CollectionRightsBackendApiService',
+      function($scope, CollectionRightsObjectFactory,
+        CollectionRightsBackendApiService) {
+        $scope.collectionId = GLOBALS.collectionId;
+        $scope.collectionRightsObject = null;
+        $scope.hasRightsLoaded = false;
+        $scope.isPrivate =
```

Comment:
The navbar and permissions tabs should be calling ``load()``, not ``fetch()``. Since JavaScript is single threaded by nature, one ``load()`` will execute first, calling a ``fetch()`` due to an empty cache. The second ``load()`` should return the same object as the first one, since the cache is warmed and does not require a second ``fetch()``.